packages/edit-site/src/components/page-content-focus-manager/edit-template-dialog.js

```diff
@@ -0,0 +1,52 @@
+/**
+ * WordPress dependencies
+ */
+import { useDispatch } from '@wordpress/data';
+import { useEffect, useState } from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+import { __experimentalConfirmDialog as ConfirmDialog } from '@wordpress/components';
+
+/**
+ * Internal dependencies
+ */
+import { store as editSiteStore } from '../../store';
+
+/**
+ * Component that displays a 'Edit your template to edit this block'
+ * notification when the user is focusing on editing page content and clicks on
+ * a disabled template block.
+ *
+ * @param {Object}                                 props
+ * @param {import('react').RefObject<HTMLElement>} props.contentRef Ref to the block
+ *                                                                  editor iframe canvas.
+ */
+export default function EditTemplateDialog( { contentRef } ) {
+	const [ isOpen, setIsOpen ] = useState( false );
+
+	useEffect( () => {
+		const handleDblClick = ( event ) => {
+			if ( event.target.classList.contains( 'is-root-container' ) ) {
+				setIsOpen( true );
+			}
+		};
+		const canvas = contentRef.current;
+		canvas?.addEventListener( 'dblclick', handleDblClick );
+		return () => canvas?.removeEventListener( 'dblclick', handleDblClick );
+	}, [ contentRef.current ] );
+
+	const { setHasPageContentFocus } = useDispatch( editSiteStore );
+
+	return (
+		<ConfirmDialog
+			isOpen={ isOpen }
+			confirmButtonText={ __( 'Edit template' ) }
+			onConfirm={ () => {
+				setIsOpen( false );
+				setHasPageContentFocus( false );
+			} }
+			onCancel={ () => setIsOpen( false ) }
+		>
+			{ __( 'Edit your template to edit this block' ) }
```

Comment:
Not a biggie for this PR, but this text can be misleading since you can click anywhere outside a content block to trigger the notice + modal, even on the latters' padding or margin.

![2023-06-16 14 11 54](https://github.com/WordPress/gutenberg/assets/6458278/6976ccc3-956a-4161-af7a-e332789d9ec6)

Maybe `"Edit your template to edit this area"` ?

Reply:
Tiny nit: for text within the dialog, should we end the sentence in a full-stop? Also, not a blocker (and wording can be tweaked in follow-ups), but since we have a modal, I was wondering if it's worth including more explanatory text in this one. E.g. something along the lines of, `"You are currently editing ${ pageTitle }, do you wish to switch to template editing?"`... though now that I write that out, I'm not sure how much more helpful that is. Writing this sort of micro copy is hard! 😅

Reply:
Will add the periods! 👍 

Not 100% sure about changing the copy. Wdyt @SaxonF?

Reply:
this sort of copy is indeed difficult. I'd be fine shipping as is and follow up with a copy pass.